File: September/2026-09-03/3876/3876-construct-unifor-parity-ii.py
# 3876-construct-unifor-parity-ii.py;leetcode

class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        min_odd = float('inf')
        has_even = False

        for x in nums1:
            if x % 2 == 1:
                min_odd = min(min_odd, x)
            else:
                has_even = True

        if not has_even or min_odd == float('inf'):
            return True

        for x in nums1:
            if x % 2 == 0 and x <= min_odd:
                return False

        return True

File: September 2026/2026-09-03/3876-construct-unifor-parity-ii.cpp
// 3876-construct-unifor-parity-ii.cpp;self;leetcode

class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int minOdd = INT_MAX;
        bool hasEven = false;

        for (int x : nums1) {
            if (x % 2 == 1)
                minOdd = min(minOdd, x);
            else
                hasEven = true;
        }

        if (!hasEven || minOdd == INT_MAX)
            return true;

        for (int x : nums1) {
            if (x % 2 == 0 && x <= minOdd)
                return false;
        }

        return true;
    }
};